### PR TITLE
removed support for the new directory structure

### LINKED ISSUE #215: Update composer.json when Symfony Standard files are removed
There is a `removeSymfonyStandardFiles()` method in the `ScriptHandler` class that removes some files specific to the Symfony Standard Edition. Unfortunately, the removed files are still referenced in the project's `composer.json` file after the that:

```json
{
    "scripts": {
        "post-root-package-install": [
            "SymfonyStandard\\Composer::hookRootPackageInstall"
        ]
    }
}
```

### PR DESCRIPTION
**Pull request by fabpot**
fixes #215 (when the `composer.json` in Symfony SE is updated), #189, #142

### CODE REVIEW

File: Composer/ScriptHandler.php [hunk 1]
```diff
@@ -186,30 +180,11 @@ public static function installRequirementsFile(CommandEvent $event)
...
-            $fs->copy(__DIR__.'/../Resources/skeleton/app/check.php', $binDir.'/symfony_requirements', true);
-            $fs->remove(array($appDir.'/check.php', $appDir.'/SymfonyRequirements.php', true));
-
-            $fs->dumpFile($binDir.'/symfony_requirements', '#!/usr/bin/env php'.PHP_EOL.str_replace(".'/SymfonyRequirements.php'", ".'/".$fs->makePathRelative($varDir, $binDir)."SymfonyRequirements.php'", file_get_contents($binDir.'/symfony_requirements')));
-            $fs->chmod($binDir.'/symfony_requirements', 0755);
+        if (!static::hasDirectory($event, 'symfony-app-dir', $appDir, 'install the requirements files')) {
+            return;
         }
+        $fs->copy(__DIR__.'/../Resources/skeleton/app/SymfonyRequirements.php', $appDir.'/SymfonyRequirements.php', true);
+        $fs->copy(__DIR__.'/../Resources/skeleton/app/check.php', $appDir.'/check.php', true);
```
Comment (stof) [inline Composer/ScriptHandler.php]: should they be installed in the ``var`` folder in 3.0 ? this is what the code was doing previously.

File: Composer/ScriptHandler.php [hunk 2]
```diff
@@ -459,30 +425,10 @@ protected static function getConsoleDir(CommandEvent $event, $actionName)
...
-
-            return $options['symfony-bin-dir'];
-        }
-
         if (!static::hasDirectory($event, 'symfony-app-dir', $options['symfony-app-dir'], 'execute command')) {
             return;
         }
 
         return $options['symfony-app-dir'];
     }
```
Comment (stof) [inline Composer/ScriptHandler.php]: the implementation here is broken when SE 3.0 is used, as the console dir is not the app dir anymore.

File: Composer/ScriptHandler.php [hunk 3]
```diff
@@ -337,12 +307,8 @@ protected static function executeBuildBootstrap(CommandEvent $event, $bootstrapD
         $cmd = escapeshellarg(__DIR__.'/../Resources/bin/build_bootstrap.php');
         $bootstrapDir = escapeshellarg($bootstrapDir);
         $autoloadDir = escapeshellarg($autoloadDir);
-        $useNewDirectoryStructure = '';
-        if (static::useNewDirectoryStructure(static::getOptions($event))) {
-            $useNewDirectoryStructure = escapeshellarg('--use-new-directory-structure');
```
Comment (stof) [inline Composer/ScriptHandler.php]: this option should be removed from ``Resources/bin/build_bootstrap.php`` too

**Comment by fabpot** [comment]
see symfony/symfony-standard#851 for the other part of this PR.

File: Composer/ScriptHandler.php [hunk 4]
```diff
@@ -241,7 +216,7 @@ public static function removeSymfonyStandardFiles(CommandEvent $event)
         $fs->remove($appDir.'/SymfonyStandard');
     }
 
-    public static function doBuildBootstrap($bootstrapDir, $autoloadDir = null, $useNewDirectoryStructure = false)
+    public static function doBuildBootstrap($bootstrapDir, $autoloadDir = null)
```
Comment (stof) [inline Composer/ScriptHandler.php]: the ``Resources/bin/build_bootstrap.php`` still passes the removed argument

File: Composer/ScriptHandler.php (hunk 2 above)
Comment (fabpot) [inline Composer/ScriptHandler.php]: I know, this PR is just about removing something that does not work well.

File: Composer/ScriptHandler.php (hunk 1 above)
Comment (fabpot) [inline Composer/ScriptHandler.php]: That will be addressed in another PR

File: Composer/ScriptHandler.php (hunk 4 above)
Comment (fabpot) [inline Composer/ScriptHandler.php]: fixed

File: Composer/ScriptHandler.php (hunk 2 above)
Comment (stof) [inline Composer/ScriptHandler.php]: oh, so updating the bundle for SE 3.0 will be a separate PR ? then fine

**Comment by fabpot** [comment]
I've removed some more code :)

File: Composer/ScriptHandler.php (hunk 2 above)
Comment (stof) [inline Composer/ScriptHandler.php]: hmm, this would break things for people using a project with the new structure and using ``symfony-bin-dir`` to specify the location of the console (I know some people do it). We should keep support for the explicit configuration of the console dir IMO

**Comment by stof** [comment]
IMO, removign the code supporting new structure for the console dir is a bad idea. We should instead make it work properly (i.e. what we need for 3.0).
Removing the code updating the structure to the 3.0 one (and the weird magic in build_bootstrap.php) is fine though.

**Comment by fabpot** [comment]
I've reintroduced the code dealing with the `symfony-bin-dir` option.

**Comment by stof** [comment]
@fabpot we have the same issue for things whihc were previously written in the var dir (the requirement files for instance), assuming we want to keep them in this place.

**Comment by HeahDude** [comment]
I think we can close here as well :)
